bec_server/bec_server/device_server/devices/device_serializer.py

```diff
@@ -186,3 +197,32 @@ def get_device_info(
             "custom_user_access": user_access,
         },
     }
+
+
+def get_devices_lazy_wait_for_connection(
+    device: PositionerBase | ComputedSignal | Signal | Device | BECDeviceBase,
+    output: dict | None = None,
+) -> dict[str, tuple[PositionerBase | ComputedSignal | Signal | Device | BECDeviceBase, bool]]:
+    """
+    Get device and subdevices of a device.
+
+    Args:
+        device (PositionerBase | ComputedSignal | Signal | Device | BECDeviceBase): device to get the devices from
+        output (dict | None): output dictionary to store the devices
+
+    Returns:
+        dict[str, tuple[PositionerBase | ComputedSignal | Signal | Device | BECDeviceBase, bool]]: Dictionary of devices with their lazy_wait_for_connection status.
+    """
+    if output is None:
+        output = {}
+    if hasattr(device, "lazy_wait_for_connection"):
+        output[device.name] = (device, device.lazy_wait_for_connection)
+    else:
+        logger.warning(f"Device {device.name} does not have lazy_wait_for_connection attribute")
+
+    for attr, cpt in device._sig_attrs.items():  # pylint: disable=protected-access
```

Comment:
probably better to use the signal / component walk methods from ophyd rather than the internal variable